### PR TITLE
chore(dataplane) rename and move failure system

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:

Move the translation failure system to its own package and rename generic components to "resource failure".

Add a config apply failure reason constant for future use.

Rename "reason" to "message", as these are both used when constructing events, and we had been calling the message "reason" in some places.

Reword some comments.

**Notes for reviewers**:

Future changes will use this system for config apply errors in addition to translation errors. Apply errors will be tracked outside the parser (in sendconfig). Those may take a while to get in because they're blocked on upstream gateway changes, so this is trying to get ahead of potential conflicts if we make other changes to the event system.

The parser instance of the collector and helper methods are still named "translation", as that instance _is_ only used for translation errors.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] ~the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR~ internal refactoring only.

### CODE REVIEW

File: test/integration/helpers_test.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func gatewayHealthCheck(ctx context.Context, client *gatewayclient.Clientset, ga
 // HTTPRouteMatchesAcceptedCallback is a testing helper functions that returns a callback which
 // checks if the provided HTTPRoute has an Accepted condition with:
 // - Status matching the provided 'accepted' boolean argument.
-// - Reason matching the provided 'reason' string argument.
+// - Message matching the provided 'reason' string argument.
```
Comment (rainest) [inline test/integration/helpers_test.go:146]:
```suggestion
// - Message matching the provided 'message' string argument.
```

File: internal/dataplane/parser/parser.go [hunk 2]
```diff
@@ -8,6 +8,10 @@ import (
 	"sort"
 	"strings"
 
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
```
Comment (randmonkey) [inline internal/dataplane/parser/parser.go:13]: `failures` package importe twice here.

File: internal/dataplane/parser/parser.go [hunk 3]
```diff
@@ -20,6 +24,7 @@ import (
 	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
```
Comment (randmonkey) [inline internal/dataplane/parser/parser.go:27]: The third time of importing `failures`. To keep the section of imports, the first 2 imports (L11~L13) should be deleted.

File: internal/dataplane/parser/ingressrules.go [hunk 4]
```diff
@@ -4,6 +4,9 @@ import (
 	"fmt"
 	"strings"
 
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
```
Comment (randmonkey) [inline internal/dataplane/parser/ingressrules.go:8]: delete the import here.

File: internal/dataplane/parser/ingressrules_test.go [hunk 5]
```diff
@@ -4,6 +4,9 @@ import (
 	"bytes"
 	"testing"
 
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
```
Comment (randmonkey) [inline internal/dataplane/parser/ingressrules_test.go:8]: delete the import here.

File: internal/dataplane/kong_client.go [hunk 6]
```diff
@@ -6,6 +6,10 @@ import (
 	"sync"
 	"time"
 
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
```
Comment (randmonkey) [inline internal/dataplane/kong_client.go:9]: delete the import here.

File: internal/dataplane/kong_client.go [hunk 7]
```diff
@@ -6,6 +6,10 @@ import (
...
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/failures"
```
Comment (randmonkey) [inline internal/dataplane/kong_client.go:11]: ditto here.

**Review by randmonkey — COMMENTED** [review]
Please check imports of `failures` in other files. Duplicate import will fail the compile, and to keep the sections of imports, only the last of them should be reserved.

File: test/integration/helpers_test.go (hunk 1 above)
Comment (czeslavo) [inline test/integration/helpers_test.go:146]:
```suggestion
// - Reason matching the provided 'reason' string argument.
```

I think this should be left without changes - Reason is the correct name for the Condition's field.

**Review by czeslavo — APPROVED** [review]

**Pull request merged**